Xamarin.Essentials/FileSystem/FileSystem.ios.cs

```diff
@@ -176,25 +179,62 @@ internal override Task<Stream> PlatformOpenReadAsync()
         }
     }
 
+    // TODO: Think if we should rename it as UIImage is just an input
     class UIImageFileResult : FileResult
     {
-        readonly UIImage uiImage;
-        NSData data;
-
-        internal UIImageFileResult(UIImage image)
+        internal UIImageFileResult(UIImage image, NSDictionary metadata)
             : base()
         {
-            uiImage = image;
-
-            FullPath = Guid.NewGuid().ToString() + FileSystem.Extensions.Png;
-            FileName = FullPath;
+            FileName = Guid.NewGuid() + FileSystem.Extensions.Jpg;
+            var tempDir = NSFileManager.DefaultManager.GetTemporaryDirectory().Path;
+            FullPath = Path.Combine(tempDir, FileName);
+            SaveImageWithMetadata(image, metadata);
         }
 
         internal override Task<Stream> PlatformOpenReadAsync()
         {
-            data ??= uiImage.AsPNG();
+            Stream fileStream = File.OpenRead(FullPath);
+
+            return Task.FromResult(fileStream);
+        }
+
+        void SaveImageWithMetadata(UIImage image, NSDictionary metadata)
+        {
+            try
+            {
+                var imageData = image.AsJPEG();
 
-            return Task.FromResult(data.AsStream());
+                if (imageData == null)
+                {
+                    // TODO: Throw custom exception
+                }
+
+                // Copy over meta data
+                using var ciImage = CIImage.FromData(imageData);
+                var originalColorSpace = ciImage.ColorSpace;
+                using var newImageSource = ciImage.CreateBySettingProperties(metadata);
+                using var ciContext = new CIContext();
+                var result = ciContext.WriteJpegRepresentation(newImageSource, NSUrl.FromFilename(FullPath), originalColorSpace ?? CGColorSpace.CreateSrgb(), new NSDictionary(), out var error);
+                if (result)
+                {
+                    if (error != null)
+                    {
+                        // TODO: Throw custom exception and include the error
+                    }
+
+                    // TODO: Throw custom exception
+                }
+            }
+            catch (Exception ex)
```

Comment:
No need for the catch here as it is fine to have the NSErrorException go out as it will contain all the info needed to debug.

Reply:
That's the whole point. I think we should better create our custom exception, so the user will know what to expect. We should add internal exceptions if the user wants to analyze exact reason.

Reply:
There is another cases, where there is no internal exception, like above in that code, where the image data might be null. Which, I believe, is a good place for custom exception as well. Something like, ImageCreationException